packages/core/src/plugins/dataTable.ts

```diff
@@ -0,0 +1,86 @@
+import cheerio from 'cheerio';
+import { FrontMatter, PluginContext } from './Plugin';
+import md from '../lib/markdown-it';
+
+const CSS_FILE_NAME = 'dataTableAssets/datatables.min.css';
+const CSS_ADDITIONAL = 'dataTableAssets/datatables-additional.css';
+const JS_FILE_NAME = 'dataTableAssets/datatables.min.js';
+
+const initScript = `
+  <script>
+    function getTableOptions(el) {
+      const options = {};
+      if ($(el).hasClass('sortable-table')) {
+        options.searching = false;
+        options.paging = false;
+        options.info = false;
+      } else if ($(el).hasClass('searchable-table')) {
+        options.ordering = false;
+        options.paging = false;
+        options.info = false;
+        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
+      } else if ($(el).hasClass('sortable-searchable-table')) {
+        options.paging = false;
+        options.info = false;
+        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
+      }
+      return options;
+    }
+
+    Vue.directive('datatable', {
+      inserted: function(el, binding) {
+        const options = binding.value || {};
+        const tableOptions = getTableOptions(el);
+        $(el).DataTable({ ...tableOptions, ...options });
+      }
+    });
+
+    document.addEventListener('DOMContentLoaded', function() {
+      $('table.sortable-table, table.searchable-table, table.sortable-searchable-table').each(function() {
+        const options = getTableOptions(this);
+        $(this).DataTable(options);
+      });
+    });
+  </script>
+`;
+
+export = {
+  getLinks: () => [
+    `<link rel="stylesheet" href="${CSS_FILE_NAME}">`,
+    `<link rel="stylesheet" href="${CSS_ADDITIONAL}">`,
+  ],
+  getScripts: () => [`<script src="${JS_FILE_NAME}"></script>`, initScript],
+  postRender: (pluginContext: PluginContext, frontmatter: FrontMatter, content: string) => {
+    const $ = cheerio.load(content);
+
+    $('m-table').each((index: number, node: cheerio.Element) => {
+      const $node = $(node);
+      const html = $node.html();
+      if (html != null) {
+        const isSortable = $node.attr('sortable') !== undefined;
+        const isSearchable = $node.attr('searchable') !== undefined;
+
+        let tableClass = 'markbind-table table table-bordered table-striped';
+        if (isSortable && isSearchable) {
+          tableClass += ' sortable-searchable-table';
+        } else if (isSortable) {
+          tableClass += ' sortable-table';
+        } else if (isSearchable) {
+          tableClass += ' searchable-table';
+        }
+
+        const renderedTable = md.render(html);
+        const $renderedTable = $(renderedTable);
+        $renderedTable.find('table')
+          .removeClass('markbind-table table table-bordered table-striped')
```

Comment:
Can i ask why did you this class in tableClass and then removed it again here? 
I think you can just add the `sortable-searchable-table`, `sortable-table` and `searchable-table` conditionally to the original class without having this step.

Reply:
Thank you for your suggestions on improving the code quality! It was originally for improving readability but is indeed unnecessary in the release code.